hyperspy/component.py

```diff
@@ -561,6 +564,32 @@ def disconnect(self, f):
         if f in self.connected_functions:
             self.connected_functions.remove(f)
 
+    def _toggle_connect_active_array(self, if_on):
+        # nothing to do (was never multidimensional)
+        if self._active_array is None:
+            return
+        # as it should be (both True)
+        if self.active_is_multidimensional and if_on:
```

Comment:
This and the next clause can be combined as `if self.active_is_multidimensional == if_on:`